rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -265,6 +275,15 @@ public Integer receiveHeader(BtcBlock header) throws IOException, BlockStoreExce
         return 0;
     }
 
+    private boolean cannotProcessNextBlock(StoredBlock previousBlock) {
+        int nextBlockHeight = previousBlock.getHeight() + 1;
+        boolean networkIsMainnet = btcContext.getParams().equals(NetworkParameters.fromID(NetworkParameters.ID_MAINNET));
+
+        return nextBlockHeight >= bridgeConstants.getBlockWithTooMuchChainWorkHeight()
+            && networkIsMainnet
```

Comment:
```suggestion
       int nextBlockHeight = previousBlock.getHeight() + 1;
        boolean networkIsMainnet = btcContext.getParams().equals(BridgeMainNetConstants.getInstance().getBtcParams());
        boolean cannotProcessBlockWithTooMuchChainWork =  networkIsMainnet && !activations.isActive(ConsensusRule.RSKIP434);
        boolean isBlockWithTooMuchChainWork = nextBlockHeight >= bridgeConstants.getBlockWithTooMuchChainWorkHeight();

        return isBlockWithTooMuchChainWork && cannotProcessBlockWithTooMuchChainWork;
```

This way looks easier to follow and understand. Wdyt?

Reply:
Btw to know which Bitcoin network is being used, you can check the bridgeConstants, and that's enough. The `bridgeSupport.btcContext` is created from the `bridgeConstants.getBtcParams`. See [here](https://github.com/rsksmart/rskj/blob/37d6b1736e0f04b3b0c4109ba511a96e91b53044/rskj-core/src/main/java/co/rsk/peg/BridgeSupportFactory.java#L67)

You can use this:

```
boolean networkIsMainnet = bridgeConstants.getBtcParams().getId().equals(NetworkParameters.ID_MAINNET);
```